mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMapper.java

```diff
@@ -57,37 +64,49 @@ public void endRequest() {
     }
 
     @Override
-    public void add(String blobId, InputStream blob) {
+    public Attachment get(AttachmentId blobId) {
+        assertIdIsNotNull(blobId);
+        ResultSet resultSet = session.execute(
+                select(BLOB)
+                    .from(TABLE_NAME)
+                    .where(eq(ID, blobId.serialize()))
+                );
+        if (resultSet.isExhausted()) {
+            throw new IllegalArgumentException("blobId");
+        } else {
+            return new SimpleAttachment(blobId, new ByteContent(resultSet.one().getBytes(BLOB).array()));
+        }
+    }
+
+    @Override
+    public void put(Attachment attachment) {
         try {
             session.execute(
                     insertInto(TABLE_NAME)
-                        .value(ID, blobId)
-                        .value(BLOB, ByteBuffer.wrap(IOUtils.toByteArray(blob)))
+                        .value(ID, getBlobId(attachment))
+                        .value(BLOB, ByteBuffer.wrap(IOUtils.toByteArray(attachment.getContent().getInputStream())))
                     );
         } catch (IOException e) {
-            e.printStackTrace();
+            throw Throwables.propagate(e);
         }
     }
 
-    @Override
-    public InputStream get(String blobId) {
-        ResultSet resultSet = session.execute(
-                select(BLOB)
-                    .from(TABLE_NAME)
-                    .where(eq(ID, blobId))
-                );
-        if (!resultSet.isExhausted()) {
-            return IOUtils.toInputStream(resultSet.one().getBytes(BLOB).asCharBuffer().toString());
-        }
-        return null;
+    private String getBlobId(Attachment attachment) {
+        return attachment.getId().serialize();
     }
 
     @Override
-    public void remove(String blobId) {
+    public void delete(AttachmentId blobId) {
+        assertIdIsNotNull(blobId);
         session.execute(
-                delete()
+                QueryBuilder.delete()
                     .from(TABLE_NAME)
-                    .where(eq(ID, blobId)));
+                    .where(eq(ID, blobId.serialize())));
```

Comment:
You can give your field the UUID type, which would be more efficient than Strings ;-)